core/templates/dev/head/collection_editor/CollectionEditor.js

```diff
@@ -25,32 +25,84 @@ oppia.constant(
-    SkillListObjectFactory, CollectionUpdateService, UndoRedoService,
+    CollectionRightsBackendApiService, ExplorationSummaryBackendApiService,
+    CollectionObjectFactory, SkillListObjectFactory,
+    CollectionValidationService, CollectionUpdateService, UndoRedoService,
     alertsService) {
     $scope.collection = null;
     $scope.collectionId = GLOBALS.collectionId;
     $scope.collectionSkillList = SkillListObjectFactory.create([]);
     $scope.isPublic = GLOBALS.isPublic;
+    $scope.validationIssues = [];
+
+    var _validateCollection = function() {
+      if ($scope.isPublic) {
+        $scope.validationIssues = (
+          CollectionValidationService.findValidationIssuesForPublicCollection(
+            $scope.collection));
+      } else {
+        $scope.validationIssues = (
+          CollectionValidationService.findValidationIssuesForPrivateCollection(
+            $scope.collection));
+      }
+    };
+
+    var _updateCollection = function(newBackendCollectionObject) {
+      $scope.collection = CollectionObjectFactory.create(
+        newBackendCollectionObject);
+      $scope.collectionSkillList.setSkills(newBackendCollectionObject.skills);
+      _validateCollection();
+    };
 
     // Load the collection to be edited.
     WritableCollectionBackendApiService.fetchWritableCollection(
       $scope.collectionId).then(
-        function(collectionBackendObject) {
-          $scope.collection = CollectionObjectFactory.create(
-            collectionBackendObject);
-          $scope.collectionSkillList.setSkills(collectionBackendObject.skills);
-        }, function(error) {
+        _updateCollection, function(error) {
           alertsService.addWarning(
-            error || 'There was an error loading the collection.');
+            error || 'There was an error when loading the collection.');
         });
 
+    UndoRedoService.setOnChangedCallback(function(changeObject, wasApplied) {
+      if (changeObject && wasApplied &&
+          CollectionUpdateService.isAddingCollectionNode(changeObject)) {
+        var explorationId = (
+          CollectionUpdateService.getExplorationIdFromChangeObject(
+            changeObject));
+        var collectionNode = (
+          $scope.collection.getCollectionNodeByExplorationId(explorationId));
+        ExplorationSummaryBackendApiService
+          .loadPublicAndPrivateExplorationSummaries(
+          [explorationId]).then(function(summaries) {
+            var summaryBackendObject = {};
+            if (summaries.length != 0 && summaries[0].id == explorationId) {
+              summaryBackendObject = summaries[0];
+              summaryBackendObject.exists = true;
+            } else {
+              summaryBackendObject.exists = false;
+            }
+            collectionNode.setExplorationSummaryObject(summaryBackendObject);
+            _validateCollection();
+          }, function() {
+            alertsService.addWarning(
```

Comment:
Should the addition be reverted, to avoid corruption of the changelist? But this is happening in a deferred process, so other changes may have come in in the meantime...

Or maybe this should be a pre-hook, before the changelist gets modified? Either way, you'd probably need a lock on the changelist.